### PR TITLE
move all routes to routes directory

### LINKED ISSUE #40: Either put all handlers in routes, or in server.js
Right now we have the most complex route handler inlined in server.js, and the one-liner route handler shuffled off into a routes/index file.

I don't care which we use (inlined in server.js is fine with me for this one, it's such a small server), but let's be consistent either way.

### PR DESCRIPTION
**Pull request by meandavejustice**
fixes #40

### CODE REVIEW

**Comment by pdehaan** [comment]
@meandavejustice Needs a rebasin'.

**Comment by meandavejustice** [comment]
@pdehaan should be good now

**Comment by pdehaan** [comment]
@meandavejustice LGTM. Only thought would be that it seems like a lot of boilerplate to create a DB connection. Not sure if you want to break that off into some ./lib/db.js file that you can share between files/routes. But considering this just a simple 1-pager, maybe I'm overthinking it.

### ./lib/db.js
```js
var AWS = require('aws-sdk');
var xtend = require('xtend');
var config = xtend(require('../app_config.json'), {
  'accessKeyId': process.env.ACCESS_KEY_ID,
  'secretAccessKey': process.env.SECRET_ACCESS_KEY
});

AWS.config.update(config);

exports.db = new AWS.DynamoDB({region: config.AWS_REGION});
```

Now, in theory, you could just simplify routes/signup.js to have 3-4 fewer `require()`s, and keeps all  your DB connection details in a single file:
```js
var db = require('../lib/db').db;
```

Or maybe even put the `signup()` method into the db.js file and abstract those details into the file. That would leave your routes/signup.js looking like:

```js
var Busboy = require('busboy');
var signup = require('./lib/db').signup;

module.exports = function(req, res) {
  try {
    var busboy = new Busboy({headers: req.headers});
    busboy.on('field', function(fieldname, val) {
      if (fieldname === 'email') {
        signup(val);
      }
    });
    busboy.on('finish', function() {
      res.sendStatus(200);
    });
    req.pipe(busboy);
  }
  catch (err) {
    res.sendStatus(500);
  }
}
```

**Comment by meandavejustice** [comment]
@pdehaan Your right, I'll pull out the db stuff.

File: server.js [hunk 1]
```diff
@@ -1,27 +1,11 @@
-//Copyright 2013-2014 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (6a68) [inline server.js]: awesome, thanks for trimming this

**Comment by 6a68** [comment]
those routes look good to me, testing now

**Comment by 6a68** [comment]
seems to work! thanks @meandavejustice :-)

**Pull request merged**